manager/integration/tests/test_recurring_job.py

```diff
@@ -476,13 +479,96 @@ def test_recurring_jobs_for_detached_volume():  # NOQA
-    17. cleanup
+        Delete the recurring backup.
+    15. In a 6-minutes retry loop, verify that the pod can eventually start.
+    16. Change the setting allow-recurring-job-while-volume-detached to false.
+    17. Cleanup.
     """
-    pass
+    recurring_job_setting = \
+        client.by_id_setting(SETTING_RECURRING_JOB_WHILE_VOLUME_DETACHED)
+    client.update(recurring_job_setting, value="true")
+
+    vol = common.create_and_check_volume(client, volume_name, size=str(1 * Gi))
+
+    lht_hostId = get_self_host_id()
+    vol.attach(hostId=lht_hostId)
+    vol = common.wait_for_volume_healthy(client, vol.name)
+
+    data = {
+        'pos': int(SIZE),
+        'content': common.generate_random_data(50 * Mi),
+    }
+    common.write_volume_data(vol, data)
+
+    vol.detach()
+    vol = common.wait_for_volume_detached(client, vol.name)
+
+    jobs = [
+        {
+            "name": RECURRING_JOB_NAME,
+            "cron": "*/1 * * * *",
+            "task": "backup",
+            "retain": 1
+        }
+    ]
+    vol.recurringUpdate(jobs=jobs)
+    common.wait_for_backup_completion(client, vol.name)
+    for _ in range(4):
+        bv = client.by_id_backupVolume(vol.name)
+        backups = bv.backupList().data
+        assert len(backups) == 1
+        time.sleep(60)
+
+    vol.recurringUpdate(jobs=[])
+
+    pv_name = volume_name + "-pv"
+    common.create_pv_for_volume(client, core_api, vol, pv_name)
+
+    pvc_name = volume_name + "-pvc"
+    common.create_pvc_for_volume(client, core_api, vol, pvc_name)
+
+    deployment_name = volume_name + "-dep"
+    deployment = make_deployment_with_pvc(deployment_name, pvc_name)
+    common.create_and_wait_deployment(apps_api, deployment)
+
+    size_mb = 400
+    pod_names = common.get_deployment_pod_names(core_api, deployment)
+    write_pod_volume_random_data(core_api, pod_names[0], "/data/test",
+                                 size_mb)
+
+    deployment['spec']['replicas'] = 0
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment["metadata"]["name"])
+
+    vol = common.wait_for_volume_detached(client, vol.name)
+
+    jobs = [
+        {
+            "name": RECURRING_JOB_NAME,
+            "cron": "*/2 * * * *",
+            "task": "backup",
+            "retain": 1
+        }
+    ]
+    vol.recurringUpdate(jobs=jobs)
+
+    common.wait_for_backup_to_start(client, vol.name)
+
+    deployment['spec']['replicas'] = 1
+    apps_api.patch_namespaced_deployment(body=deployment,
+                                         namespace='default',
+                                         name=deployment["metadata"]["name"])
+
+    deployment_label_name = deployment["metadata"]["labels"]["name"]
+    common.wait_pod_auto_attach_after_first_backup_completion(
+        client, core_api, vol.name, deployment_label_name)
+
```

Comment:
Can we put 
```
time.sleep(30)
```
here? After the recurring job finish, we need to give it sometime so that it can detach the volume properly. Deleting it immediately will make it not be able to detach the volume and the consuming workload can never use the volume which makes the test case fail.

Reply:
Addressed. The wait time looks random though, I wonder if there are ways to check when recurring job can be deleted?

Reply:
Good idea! We can wait for the volume to be in (detached state or frontend is enabled). It means that the recurring job has finished and thus we can delete it

Reply:
Addressed. I've removed the sleep and check for volume detach/attach and volume frontend instead. Thanks :)

Reply:
Thank you!